SPIR-V Environment Spec Refactor

bashbaug opened the pull request:
This change refactors the SPIR-V environment spec to increase readability, reduce duplicated information, and break the tight coupling between an OpenCL version and the SPIR-V version(s) it supports.

I've fixed a few bugs as well, see #80 and #81.

kpet commented on env/common_properties.asciidoc:
```diff
@@ -196,18 +154,18 @@ The following table describes other data types that may be used in an
 // clk_event_t
 | *OpTypeDeviceEvent*
-| OpenCL device-side event type.
+| OpenCL device-side event representing commands enqueued to a device command queues.
```
typo

bashbaug replied:
Fixed, thanks!

kpet commented on env/introduction.asciidoc:
```diff
@@ -0,0 +1,29 @@
+language capable of representing parallel compute kernels that are executed
+by implementations of the OpenCL standard.
+
+SPIR-V is adaptable to multiple execution environments: A SPIR-V module is
```
s/A/a/

bashbaug replied:
Fixed, thanks!

kpet commented on env/introduction.asciidoc:
```diff
@@ -0,0 +1,29 @@
+
+This document is written for compiler developers who are generating SPIR-V
+modules intended to be consumed by the OpenCL API, for implementors of the
+OpenCL API who are consuming SPIR-V modules, and by software developers who
```
Did you intend to write "by"?

bashbaug replied:
Fixed: by -> for.

bashbaug commented:
Thanks @kpet, I believe I've addressed all of your comments.  Note, I've opened #88 regarding one part of the 64-bit atomics section, but I don't think that should hold up these changes.

kpet commented:
Thanks! Well-spotted.Yes, it shouldn't hold up this PR (commenting on the Int64Atomics capability was already a bit of a stretch).
